lib/std/heap.zig

```diff
@@ -8,6 +8,129 @@ const c = std.c;
 const Allocator = std.mem.Allocator;
 const windows = std.os.windows;
 
+const page_size_darwin: ?comptime_int = if (builtin.os.tag.isDarwin())
+    switch (builtin.cpu.arch) {
+        .x86, .x86_64 => 4 << 10,
+        .thumb, .thumbeb, .arm, .armeb, .aarch64, .aarch64_be, .aarch64_32 => 16 << 10,
```

Comment:
It would be good to include sources for these if possible (I know they weren't there before either).

Reply:
Please remove the 32-bit Arm tags from this (they're about as dead as macOS on 32-bit x86). Other than that, these look correct.

This would be good to link to as a source: https://developer.apple.com/library/archive/documentation/Performance/Conceptual/ManagingMemory/Articles/AboutMemory.html ("About Virtual Memory" section)